crates/verify/src/etherscan/flatten.rs

```diff
@@ -190,7 +190,7 @@ Diagnostics: {diags}",
             include_paths: Default::default(),
         };
 
-        let out = zksolc.compile(&input)?;
+        let out = zksolc.compile(&input.input)?;
```

Comment:
we don't support VersionedInput?

Reply:
We do but there's a `ZkSolcCompiler` data structure that handles it, one level above the `ZkSolc` one: https://github.com/Moonsong-Labs/compilers/blob/b597d1d8887e2568dda08f81c45d3ec2df832a46/crates/compilers/src/compilers/zksolc/mod.rs#L153

ZkSolc now represents the raw command (they did this for solc as well)

Reply:
Then we should use that, to preserve the same semantics

Reply:
@Karrq modified. Also moved `ZkVersion` to the context so the appropriate  zksolc/solc versions are used and the flow is more clear